SAPHana/ra/SAPHanaController

```diff
@@ -2596,6 +2648,20 @@ function saphana_monitor_primary()
                 ;;
             # TODO: PRIO5: Implement WAITING4LPA WAITING4REG WAITING4PRIM WAITING4NODES
             # WAITING4LPA WAITING4REG WAITING4PRIM WAITING4NODES
+            # TODO: PRIO1: Implement WAITING4NODES in the same way as for secondary:
+#        1 ) # ERROR
+#            super_ocf_log debug "DBG: 012 * lpa_set_lpt 10 $sr_name"
+#            lpa_set_lpt  10 "$sr_name"
+#            # Hide this error, if we still in WAITING4NODES
+#            case "$promote_attr" in
+#                WAITING4NODES ) # hide error 
+#                                rc=$OCF_SUCCESS
```

Comment:
In order to make it clear for users, we could probably return the value `190` here, which is co-called `OCF_DEGRADED`:

https://clusterlabs.org/pacemaker/doc/2.1/Pacemaker_Administration/html/agents.html#ocf-return-codes

It basically means the status of the resource instance is not perfect but still acceptable. Pacemaker won't do anything about it but only show something like the following from UI like crm_mon:

`rsc_SAPHanaCon_HA1_HDB10_monitor_61000 on suse11 'OCF_DEGRADED' (190): call=99, status='complete', last-rc-change='Thu Nov 24 15:46:40 2022', queued=0ms, exec=9241ms`

Although the return codes 190 (`OCF_DEGRADED`)  and 191 (`OCF_DEGRADED_PROMOTED`) have become official only in OCF 1.1

https://github.com/ClusterLabs/OCF-spec/blob/main/ra/1.1/resource-agent-api.md#exit-status-codes 

,  but they have been supported by pacemaker since even SLE12 SP1.

We could directly use the return code value 190 or 191 until they are formally defined in ocf-shellfuncs/ocf-returncodes of resource-agents.

Reply:
Thanks, @gao-yan .